OutOfSchool/OutOfSchool.WebApi.Tests/Extensions/MapSearchResultToOkOrNoContentExtensionTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections.Generic;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using NUnit.Framework;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.BusinessLogic.Models.Application;
+using OutOfSchool.WebApi.Extensions;
+
+namespace OutOfSchool.WebApi.Tests.Extensions;
+
+[TestFixture]
+public class MapSearchResultToOkOrNoContentExtensionTests
+{
+    private class TestController : ControllerBase
+    {
+    }
+
+    private TestController controller;
+
+    [SetUp]
+    public void SetUp()
+    {
+        controller = new TestController();
+    }
+
+    [Test]
+    public void MapSearchResultToOkOrNoContentExtension_SearchResultIsNotNullAndNotEmpty_ReturnOk()
+    {
+        // Arrange
+        var applications = new List<ApplicationDto>() {
+            new ApplicationDto() { RejectionMessage = "Test 1" },
+            new ApplicationDto() { RejectionMessage = "Test 2", },
+            new ApplicationDto() { RejectionMessage = "Test 3" },
+        };
+
+        var searchResult = new SearchResult<ApplicationDto>() { TotalAmount = applications.Count, Entities = applications };
+
+        // Act
+        var result = controller.MapSearchResultToOkOrNoContent(searchResult);
+
+        // Assert
+        result.Should().NotBeNull();
+        result.Should()
+              .BeOfType<OkObjectResult>()
+              .Which.StatusCode
+              .Should()
+              .Be(StatusCodes.Status200OK);
```

Comment:
(Test assertions) Please don't test dependencies, test only SUT code.
Here you check that result is of `OkObjectResult` type. Is this type generation part of our code? No, it's generated in `controller.Ok()` that is not SUT and thus should be mocked. Instead, you should check that `Ok()` was called, but `NoContent()` - wasn't. See example
```
        // Arrange
        var applications = new Faker<ApplicationDto>().Generate(3);
        var searchResult = new SearchResult<ApplicationDto>() { Entities = applications };
        var expectedResult = new OkObjectResult(new { });

        var mockedController = new Mock<ControllerBase>();
        mockedController.Setup(c => c.Ok(searchResult)).Returns(expectedResult);

        // Act
        var result = mockedController.Object.MapSearchResultToOkOrNoContent(searchResult);

        // Assert
        result.Should().Be(expectedResult);

        mockedController.VerifyAll();
        mockedController.Verify(c => c.Ok(searchResult), Times.Once);
        mockedController.Verify(c => c.NoContent(), Times.Never);
```
Here we check that result of `Ok()` is passed without any transformation outside, `Ok()` is called once? `NoContent()` is never called, and (by using `.VerifyAll()`) that not any other controller method is called.